File: r-package/R/read_pop_arrangements.R
#' Download population arrangements in Brazil
#'
#' @description
#' This function reads the official data on the population arrangements (Arranjos
#' Populacionais) in Brazil. Original data by the Brazilian Institute of Geography
#' and Statistics (IBGE). More information about the methodology at \url{https://www.ibge.gov.br/apps/arranjos_populacionais/2015/pdf/publicacao.pdf}
#'
#' @template year
#' @template code_state
#' @template simplified
#' @template output
#' @template showProgress
#' @template cache
#' @template verbose
#'
#' @return An `"sf" "data.frame"` OR an `ArrowObject`
#'
#' @export
#'
#' @examplesIf identical(tolower(Sys.getenv("NOT_CRAN")), "true")
#' # Read population arrangements in an specific year
#' pa <- read_pop_arrangements(year = 2010)
#'
read_pop_arrangements <- function(year,
                                  code_state = "all",
                                  simplified = TRUE,
                                  output = "sf",
                                  showProgress = TRUE,
                                  cache = TRUE,
                                  verbose = TRUE){

  # Get metadata with data url addresses
  temp_meta <- select_metadata(
    geography="poparrangements",
    year = year,
    simplified = simplified,
    verbose = verbose
  )

  # download files
  temp_arrw <- download_parquet(
    filename_to_download = temp_meta$file_name,
    showProgress = showProgress,
    cache = cache
  )

  # check if download failed
  if (is.null(temp_arrw)) { return(invisible(NULL)) }

  # keep only pop_arrangement
  temp_arrw <- temp_arrw |>
    dplyr::filter(! is.na(code_pop_arrangement))

  # FILTER
  temp_arrw <- filter_arrw(temp_arrw, code = code_state)

  # convert to sf
  temp <- convert_arrow2sf(temp_arrw, output)

  return(temp)
}

File: python-package/geobr/read_pop_arrangements.py
from geobr.utils import read_geobr_v2
from geobr._output import convert_output
from geobr._duckdb_backend import duckdb_connection


def read_pop_arrangements(
    year: InterruptedError,
    code_state: str = "all",
    simplified: bool = True,
    verbose: bool = False,
    output: str = "gpd",
    show_progress: bool = True,
    cache: bool = True,
):
    """Download population arrangements (IBGE).

    Parameters
    ----------
    year : int
        Year of the data (2010 in v2).
    code_state : str or int
        State abbrev, two-digit code, or ``"all"``.
    simplified, verbose, output, show_progress, cache
        Standard geobr options.
    """
    relation = read_geobr_v2(
        "poparrangements",
        year,
        code=code_state,
        simplified=simplified,
        output="duckdb",
        show_progress=show_progress,
        cache=cache,
        verbose=verbose,
    )

    conn = duckdb_connection()

    relation = conn.sql(
        "SELECT * FROM relation WHERE code_pop_arrangement IS NOT NULL"
    )

    return convert_output(relation, output, conn)
